op-e2e/system_test.go

```diff
@@ -155,6 +156,71 @@ func TestL2OutputSubmitter(t *testing.T) {
 	}
 }
 
+func TestSystemE2EDencunAtGenesis(t *testing.T) {
+	InitParallel(t)
+
+	cfg := DefaultSystemConfig(t)
+	genesisActivation := uint64(0)
+	cfg.DeployConfig.L1CancunTimeOffset = &genesisActivation
+
+	sys, err := cfg.Start(t)
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+	runE2ESystemTest(t, sys)
+	head, err := sys.Clients["l1"].BlockByNumber(context.Background(), big.NewInt(0))
+	require.NoError(t, err)
+	require.NotNil(t, head.ExcessBlobGas(), "L1 is building dencun blocks since genesis")
+}
+
+// TestSystemE2EDencunAtGenesis tests if L2 finalizes when blobs are present on L1
+func TestSystemE2EDencunAtGenesisWithBlobs(t *testing.T) {
+	InitParallel(t)
+
+	cfg := DefaultSystemConfig(t)
+	//cancun is on from genesis:
+	genesisActivation := uint64(0)
+	cfg.DeployConfig.L1CancunTimeOffset = &genesisActivation // i.e. turn cancun on at genesis time + 0
+
+	sys, err := cfg.Start(t)
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+
+	// send a blob-containing txn on l1
+	ethPrivKey := sys.cfg.Secrets.Alice
+	txData := transactions.CreateEmptyBlobTx(ethPrivKey, true)
+	txData.ChainID.SetUint64(uint64(sys.cfg.L1ChainIDBig().Uint64())) // convert from big.int chainid to uint256 chainid
+	tx := types.MustSignNewTx(ethPrivKey, types.LatestSignerForChainID(cfg.L1ChainIDBig()), txData)
+	// send blob-containing txn
+	sendCtx, sendCancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer sendCancel()
+
+	require.True(t, tx.GasTipCap().Cmp(tx.GasFeeCap()) > 0, "GasTipCap should be greater than GasFeeCap")
+	require.True(t, tx.GasTipCap().Cmp(tx.BlobGasFeeCap()) > 0, "GasTipCap should be greater than BlobFeeCap")
+
+	l1Client := sys.Clients["l1"]
+	err = l1Client.SendTransaction(sendCtx, tx)
+	require.NoError(t, err, "Sending L1 empty blob tx")
```

Comment:
It's interesting, this `require.NoError` fails with `max priority fee per gas higher than max fee per gas` but this doesn't seem to be true (per the `require.True` gas checks above, which pass)